### PR TITLE
Adding new option to reverse the priorities on delete

### PR DESCRIPTION
**Pull request by luisdavim**

### CODE REVIEW

**Comment by luisdavim** [comment]
@bergerx because otherwize I would need to create a new structure for the settings, it mapped to `map[string]string`

**Comment by bergerx** [comment]
Oh I see, it's coming from
https://github.com/Praqma/helmsman/blob/71e1165ce72d87536868a3134d291bb5bfe80e46/state.go#L27

Extending the state.go to address this doesn't seem hard, but could create
some unexpected overhead on this PR since you have to touch multiple
unrelated places which uses the settings map.

Sorry if I caused unnecessary overhead.

**Comment by luisdavim** [comment]
Yes, no worries, I'll look into it tomorrow.

**Comment by sami-alajrami** [comment]
boolean would be more consistent with the other options. A simple solution to avoid the overhead of modifying the settings map is to parse the string into boolean .. something like: `strconv.ParseBool()` which also requires validating/handling of none boolean string values.

**Comment by luisdavim** [comment]
I've created a new struct for the settings, please revire @bergerx and @sami-alajrami

File: state.go [hunk 1]
```diff
@@ -20,11 +20,23 @@ type namespace struct {
 	ClientKey            string `yaml:"clientKey"`
 }
 
+// config type represents the settings fields
+type config struct {
```
Comment (bergerx) [inline state.go]: Why the struct name is `config` and not `settings`?

Comment (luisdavim) [reply state.go]: It was to make sure it wasn't going to conflict with any other existing variable. I can change it but, it is that important?

**Comment by luisdavim** [comment]
Thanks for reviewing so quickly @bergerx .

File: state.go (hunk 1 above)
Comment (bergerx) [reply state.go]: Not at all, that fine as is.

**Comment by sami-alajrami** [comment]
LGTM. Thanks @luisdavim !

**Pull request merged**